holoviews/operation/datashader.py

```diff
@@ -139,17 +139,28 @@ def get_agg_data(cls, obj, category=None):
                     df[obj.vdims[0].name] = p.level
                 paths.append(df)
         elif isinstance(obj, CompositeOverlay):
+            element = None
             for key, el in obj.data.items():
                 x, y, element, glyph = cls.get_agg_data(el)
+                dims = (x, y)
                 df = PandasInterface.as_dframe(element)
                 if isinstance(obj, NdOverlay):
                     df = df.assign(**dict(zip(obj.dimensions('key', True), key)))
                 paths.append(df)
-            kdims += element.kdims
-            vdims = element.vdims
+            if element is None:
+                dims = None
+            else:
+                kdims += element.kdims
+                vdims = element.vdims
         elif isinstance(obj, Element):
             glyph = 'line' if isinstance(obj, Curve) else 'points'
             paths.append(PandasInterface.as_dframe(obj))
+
+        if dims is None or len(dims) != 2:
+            return None, None, None, None
```

Comment:
Purely syntactic preference, but I would wrap the return tuple in parentheses i.e ``(None, None, None, None)``